backend/UserApp/views.py

```diff
@@ -150,36 +152,33 @@ def put(self, request, username):
 
 
 class UserUpdatePasswordView(APIView):
-    def put(self, request, username):
+    def put(self, request):
+        email = request.data.get('email', None)
+        old_password = request.data.get('old_password', None)
+        new_password = request.data.get('new_password', None)
+
+        if not email or not old_password or not new_password:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        # Find user by email
         try:
-            user = User.objects.get(username=username)
+            user = User.objects.get(email=email)
         except User.DoesNotExist:
             return Response(
-                {'error': 'User not found'},
                 status=status.HTTP_404_NOT_FOUND
```

Comment:
I know that we recently wanted to change the response format to just have a status associated with it, but Polina understandably wants some error logging to help her debug how to use the APIs. 

I would attach an error field with a message saying what happened during your request to your error responses so the frontend people have an easier time debugging what they're missing in the request.